lib/raxx/router.ex

```diff
@@ -1,34 +1,62 @@
 defmodule Raxx.Router do
   @moduledoc """
-  Simple router for Raxx applications.
+  Routing for Raxx applications.
 
-  A router is a list of associations between a request pattern and controller module.
-  `Raxx.Router` needs to be used after `Raxx.Server`,
-  it is an extension to a standard Raxx.Server.
-
-  Each controller module that is passed requests from `Raxx.Router` are also standalone `Raxx.Server`s.
-
-  *`Raxx.Router` is a deliberatly low level interface that can act as a base for more sophisticated routers.
-  `Raxx.Blueprint` part of the [tokumei project](https://hexdocs.pm/tokumei/Raxx.Blueprint.html) is one example.
+  Routes are defined as a match and an action module.
+  Standard Elixir pattern matching is used to apply the match to an incoming request.
+  An action module another implementation of `Raxx.Server`
 
+  Sections group routes that all have the same middleware.
+  Middleware in a section maybe defined as a list,
+  this is useful when all configuration is known at compile-time.
+  Alternativly an arity 1 function can be used.
+  This can be used when middleware require runtime configuration.
+  The argument passed to this function is server initial state.
 
   ## Examples
 
       defmodule MyRouter do
-        use Raxx.Server
+        use Raxx.Router
 
-        use Raxx.Router, [
+        section [{Raxx.Logger, level: :debug}], [
+          {%{method: :GET, path: ["ping"]}, Ping},
+        ]
+
+        section &web/1, [
           {%{method: :GET, path: []}, HomePage},
           {%{method: :GET, path: ["users"]}, UsersPage},
           {%{method: :GET, path: ["users", _id]}, UserPage},
           {%{method: :POST, path: ["users"]}, CreateUser},
           {_, NotFoundPage}
         ]
+
+        def web(state) do
+          [
+            {Raxx.Logger, level: state.log_level},
+            {MyMiddleware, foo: state.foo}
+          ]
+        end
       end
+
+  *The original API is kept for backwards compatibility.
+  See [previous docs](https://hexdocs.pm/raxx/0.17.2/Raxx.Router.html) for details.*
+
+  *If the sections DSL does not work for an application it is possible to instead just implement a `route/2` function.*
```

Comment:
From what I'm reading it will work (as long as there's no routes defined using the DSL), but it would be really good to have a taste case for doing this (writing your own route function)